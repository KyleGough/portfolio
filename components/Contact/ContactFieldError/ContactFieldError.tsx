import { clsx } from 'clsx';
import React from 'react';

interface ContactFieldErrorProps {
  message: string;
}

export const ContactFieldError: React.FC<ContactFieldErrorProps> = ({
  message,
}) => (
  <span
    className={clsx(
      'transition-opactity',
      'duration-200',
      'text-sm',
      'text-error'
    )}
  >
    {message}
  </span>
);
