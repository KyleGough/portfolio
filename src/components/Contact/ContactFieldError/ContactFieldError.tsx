import { clsx } from 'clsx';
import React from 'react';

interface ContactFieldErrorProps {
  show: boolean;
  message: string;
}

export const ContactFieldError: React.FC<ContactFieldErrorProps> = ({
  show,
  message,
}) => (
  <span
    className={clsx(
      { hidden: !show },
      'transition-opactity',
      'duration-200',
      'text-sm',
      'text-error'
    )}
  >
    {message}
  </span>
);
