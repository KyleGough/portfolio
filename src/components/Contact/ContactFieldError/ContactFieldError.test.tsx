import { render, screen } from '@testing-library/react';
import React from 'react';

import { ContactFieldError } from './ContactFieldError';

describe('ContactFieldError component', () => {
  it('renders', async () => {
    const { container } = render(
      <ContactFieldError message="Invalid Email!" />
    );

    expect(screen.getByText('Invalid Email!')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
