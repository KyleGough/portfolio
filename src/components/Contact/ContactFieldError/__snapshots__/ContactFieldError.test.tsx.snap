// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactFieldError component renders 1`] = `
<div>
  <span
    class="transition-opactity duration-200 text-sm text-error"
  >
    Invalid Email!
  </span>
</div>
`;
